Enabled usetesting linter in golangci-lint

Yadavanurag13 opened the pull request:
This PR closes #1612 


**Description:**

-   Enabled usetesting

**Checklist:**

-   [X] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [ ] All new code is covered by unit tests.
-   [X] This PR does not decrease the overall code coverage.
-   [X] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

ccoVeille commented on .golangci.yml:
```diff
@@ -147,6 +147,40 @@ linters:
     - wsl
+    - usetesting
+  
+  settings:
+      usetesting:
+        # Enable/disable `os.CreateTemp("", ...)` detections.
+        # Default: true
+        os-create-temp: false
+    
+        # Enable/disable `os.MkdirTemp()` detections.
+        # Default: true
+        os-mkdir-temp: false
+    
+        # Enable/disable `os.Setenv()` detections.
+        # Default: true
+        os-setenv: false
+    
+        # Enable/disable `os.TempDir()` detections.
+        # Default: false
+        os-temp-dir: true
+        
+        # Enable/disable `os.Chdir()` detections.
+        # Disabled if Go < 1.24.
+        # Default: true
+        os-chdir: false
+    
+        # Enable/disable `context.Background()` detections.
+        # Disabled if Go < 1.24.
+        # Default: true
+        context-background: false
+    
+        # Enable/disable `context.TODO()` detections.
+        # Disabled if Go < 1.24.
+        # Default: true
+        context-todo: false
```
Please remove this that is not used.

The config is based on golangci-lint  v1, so the node for this would be linter-settings.usetesting not `settings.usetesting`

Also, you are disabling all usetesting configuration by reverting every single setting to the opposite of their default value


```suggestion
```

Also, you are breaking the comment about the linters that is disabled on purpose like godot

Yadavanurag13 replied:
I have removed these changes.

ccoVeille commented on go.work.sum:
I don't understand how this file could be modified without having a change in go.mod

Maybe the go.mod/go.sum is not clean, but then it should be a separate PR

My guess is this was altered during your local tests and was committed by error

Yadavanurag13 replied:
I have rebased my pr on latest development branch. may be there I have committed it by err.

Fixing it.

ccoVeille reviewed (commented):
LGTM 👍 except minors remarks

Yadavanurag13 commented:
@ccoVeille I have pushed the final changes and addressed all the comments. However, there's still a diff showing in the go.work.sum file, even though the contents are exactly the same. Could you please review it once?

Thanks !!
